starter-code/clue.js

```diff
@@ -1,104 +1,256 @@
-Observatory
-Theater
-Guest House
-Patio
+  image: "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+  occupation: "Actor"
+};
 
-*/
+var mrsPeacock = {
+  firstName: "Eleanor",
+  lastName: "Peacock",
+  color: "blue",
+  description: "She is from a wealthy family and uses her status and money to earn popularity",
+  age: 36,
+  image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+  occupation: "Socialité"
+};
 
-function random_selector(array) {
+ var mrMustard = {
+  firstName: "Jack",
+  lastName: "Mustard",
+  color: "yellow",
+  description: "He is a former football player who tries to get by on his former glory",
+  age: 62,
+  image: "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+  occupation: "Retired Football player"
+};
+
+// ---- Weapons ----
+
+var rope = {
+  name: "Rope",
+  weight: 10,
+  damage: 20
+};
+
+var knife = {
+  name: "Knife",
+  weight: 8,
+  damage: 16
+};
+
+var candleStick = {
+  name: "Candlestick",
+  weight: 2,
+  damage: 4
+};
+
+var dumbBell = {
+  name: "Dumbbell",
+  weight: 30,
+  damage: 60
+};
+
+
+var poison = {
+  name: "Poison",
+  weight: 2,
+  damage: 4
+};
+
+var axe = {
+  name: "Axe",
+  weight: 15,
+  damage: 30
+};
+
+var bat = {
+  name: "Bat",
+  weight: 13,
+  damage: 26
+};
+
+var trophy = {
+  name: "Trophy",
+  weight: 25,
+  damage: 50
+};
+
+var pistol = {
+  name: "Pistol",
+  weight: 20,
+  damage: 40
+};
+
+//---- Rooms ----
+
+var diningRoom = {
+  name: "Dining Room"
+};
+
+var conservatory = {
+  name: "Conservatory"
+};
+
+var kitchen = {
+  name: "Kitchen"
+};
+
+var study = {
+  name: "Study"
+};
 
+var library = {
+  name: "Library"
+};
+
+var billiardRoom = {
+  name: "Billiard Room"
+};
+
+var lounge = {
+  name: "Lounge"
+};
+
+var ballRoom = {
+  name: "Ballroom"
+};
+
+var hall = {
+  name: "Hall"
+};
+
+var spa = {
+  name: "Spa"
+};
+
+var livingRoom = {
+  name: "Living Room"
+};
+
+var observatory = {
+  name: "Observatory"
+};
+
+var theater = {
+  name: "Theater"
+};
+
+var guestHouse = {
+  name: "Guest House"
+};
+
+var patio = {
+  name: "Patio"
+};
+
+//----- functions
+
+ var rooms= [diningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballRoom, hall, spa, livingRoom, observatory, theater, guestHouse, patio];
+ var characters = [mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard];
+ var weapons = [rope, knife, candleStick, dumbBell, poison, axe, bat, trophy, pistol];
+/*
+function random_selector(array) {
+  var rnd = Math.random();
+  return array[Math.floor(rnd * (array.length))];
+}
+*/
+function random_selector(array) {
+  var rnd = Math.random();
+  var newIndex = Math.floor(rnd * (array.length));
+  var element = array[newIndex];
+  array.splice(newIndex, 1);
+  return element;
 }
 
 function pick_mistery() {
-
+  var room = random_selector(rooms);
+  var weapon = random_selector(weapons);
+  var character = random_selector(characters);
+  var mistery = {
```

Comment:
You don't need the intermediate variables, but you can keep them if you think it makes the code easier to understand. If you don't want them you directly write:

```
var mistery = {
  room: random_selector(rooms),
  weapon: random_selector(weaponss),
  character: random_selector(characters),
}
```